### PR TITLE
Removed -f option from java_home for macOS Big Sur

### PR DESCRIPTION
**Pull request by catap**
#### Description

macOS Big Sur hasn't got anymore -f option for /usr/libexec/java_home and this
optinon brokes it on macOS Big Sur.

As soon as I removed it the behaviour is going back to expected.

Closes: https://trac.macports.org/ticket/61445

It also reverted a5935f1fd89e02a43832bd9914766729b76ca792

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 11.0.1 20B50
Xcode 12.2 12B45b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: _resources/port1.0/group/java-1.0.tcl [hunk 1]
```diff
@@ -58,14 +47,19 @@ proc find_java_home {} {
 
     global java.version java.fallback
     if { ${java.version} ne "" } {
-        if { [catch {set val [exec "/usr/libexec/java_home" "-f" "-v" ${java.version}]}] } {
+        set f_param "-f"
+        global os.platform os.major
+        if {${os.platform} eq "darwin" && ${os.major} == 20} {
```
Comment (cjones051073) [inline _resources/port1.0/group/java-1.0.tcl:52]: Probably better to make that check darwin >= 20 to cover future versions, intil we know when/if apple fixes the bug.

**Comment by cjones051073** [comment]
I don't see the `-f` option on older macOS versions either. e.g. on macOS 10.13

```
Titan ~ > /usr/libexec/java_home -h
Usage: java_home [options...]
    Returns the path to a Java home directory from the current user's settings.

Options:
    [-v/--version   <version>]       Filter Java versions in the "JVMVersion" form 1.X(+ or *).
    [-a/--arch      <architecture>]  Filter JVMs matching architecture (i386, x86_64, etc).
    [-d/--datamodel <datamodel>]     Filter JVMs capable of -d32 or -d64
    [-t/--task      <task>]          Use the JVM list for a specific task (Applets, WebStart, BundledApp, JNI, or CommandLine)
    [-F/--failfast]                  Fail when filters return no JVMs, do not continue with default.
    [   --exec      <command> ...]   Execute the $JAVA_HOME/bin/<command> with the remaining arguments.
    [-R/--request]                   Request installation of a Java Runtime if not installed.
    [-X/--xml]                       Print full JVM list and additional data as XML plist.
    [-V/--verbose]                   Print full JVM list with architectures.
    [-h/--help]                      This usage information.
```

I suspect the intention is to use the `-F` option, and perhaps using lower case is accepted. 

In any case, I suggest you switch the option to use `-F` just to match what the help for the command lists.

Also note, this option is still broken on macOS 11

```
Oberon ~/Projects/MacPorts/ports > /usr/libexec/java_home -h
Usage: java_home [options...]
    Returns the path to a Java home directory from the current user's settings.

Options:
    [-v/--version   <version>]       Filter versions (as if JAVA_VERSION had been set in the environment).
    [-a/--arch      <architecture>]  Filter architecture (as if JAVA_ARCH had been set in the environment).
    [-F/--failfast]                  Fail when filters return no JVMs, do not continue with default.
    [   --exec      <command> ...]   Execute the $JAVA_HOME/bin/<command> with the remaining arguments.
    [-X/--xml]                       Print full JVM list and additional data as XML plist.
    [-V/--verbose]                   Print full JVM list with architectures.
    [-h/--help]                      This usage information.
Oberon ~/Projects/MacPorts/ports > /usr/libexec/java_home -v 1.8+
/Library/Java/JavaVirtualMachines/openjdk15/Contents/Home
Oberon ~/Projects/MacPorts/ports > /usr/libexec/java_home -F -v 1.8+
The operation couldn’t be completed. Unable to locate a Java Runtime that supports (null).
Please visit http://www.java.com for information on installing Java.
```

so appears the workaround in this PR is still required, at least for now.

**Comment by cjones051073** [comment]
Actually.... Thinking a bit more I think this PR should not be merged, as removing the -F option means the command will return  the default JDK on failure to find a match. e.g.

```
Oberon ~/Projects/MacPorts/ports > /usr/libexec/java_home -v 16+    
/Library/Java/JavaVirtualMachines/openjdk15/Contents/Home
```

this is not what we want, so I am afraid I don't think the PR show be applied...

**Comment by catap** [comment]
@cjones051073 maybe at least fix a typo?

**Pull request closed without merging**